flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/batch/sql/CalcTest.scala

```diff
@@ -187,4 +187,13 @@ class CalcTest extends TableTestBase {
   def testDecimalMapWithDifferentPrecision(): Unit = {
     util.verifyExecPlan("SELECT MAP['a', 0.12, 'b', 0.5]")
   }
+
+  @Test
+  def testCastMapValueInsideArray(): Unit = {
+    util.verifyExecPlan(
```

Comment:
Can we also add a test where the key or value (or both) is of some other type, like a numeric?